app/components/Button.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { ButtonProps, TouchableOpacity } from 'react-native'
+import { tailwind } from '../tailwind'
+import { Text } from './Text'
+
+export type ButtonFillTypes = 'fill' | 'outline' | 'flat'
+export type ButtonColor = 'primary' | 'secondary'
+
+export function Button (props: React.PropsWithChildren<ButtonProps> & { color?: ButtonColor, fill?: ButtonFillTypes, label?: string }): JSX.Element {
+  const buttonColor = props.color ?? 'primary'
+  const buttonFill = props.fill ?? 'fill'
```

Comment:
```suggestion
  const { 
    color = 'primary',
    fill = 'fill',
    ...
  } = props
```

This kind of default restriction might be easier to grasps